crates/redis_interface/src/commands.rs

```diff
@@ -147,13 +147,17 @@ impl super::RedisConnectionPool {
             .change_context(errors::RedisError::JsonDeserializationFailed)
     }
 
+   
+
     #[instrument(level = "DEBUG", skip(self))]
     pub async fn delete_key(&self, key: &str) -> CustomResult<(), errors::RedisError> {
-        self.pool
+        match self.pool
             .del(key)
-            .await
-            .into_report()
-            .change_context(errors::RedisError::DeleteFailed)
+            .await{
+                Ok(0) | Ok(i32::MIN..=-1_i32) => Err(errors::RedisError::NotFound).into_report(),
+                Ok(1_i32..=i32::MAX) =>  Ok(()),
+                Err(fred::error::RedisError {..}) => return Err(errors::RedisError::DeleteFailed).into_report(),              
```

Comment:
We'd prefer matching against 0 and 1 only, while keeping the other values as unexpected responses. In the previous conversation, I forgot that we were doing similar response handling elsewhere in the `redis_interface` crate, in a slightly neater way. Could you define an enum `DelReply` like so, which implements the `fred::types::FromRedis` trait:

```rust
enum DelReply {
    KeyDeleted,
    KeyNotDeleted, // Key not found
}
```

You can refer to this snippet for implementing the `FromRedis` trait:

https://github.com/juspay/hyperswitch/blob/d7cfb4a179083580a7e195fa07077af23a262ceb/crates/redis_interface/src/types.rs#L135-L152

Then, you can match on the `DelReply` enum and return either an `Ok(())`, `Err(RedisError::NotFound)`, or `Err(RedisError::DeleteFailed)`.

Reply:
done